cmd/kubectl-gadget/seccompadvisor.go

```diff
@@ -0,0 +1,188 @@
+}
+
+var seccompAdvisorListCmd = &cobra.Command{
+	Use:          "list",
+	Short:        "List existing seccomp traces",
+	RunE:         runSeccompAdvisorList,
+	SilenceUsage: true,
+}
+
+var (
+	outputMode         string
+	seccompProfileName string
+)
+
+func init() {
+	// Add generic information.
+	rootCmd.AddCommand(seccompAdvisorCmd)
+	utils.AddCommonFlags(seccompAdvisorCmd, &params)
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorStartCmd)
+	seccompAdvisorStartCmd.PersistentFlags().StringVarP(&outputMode,
+		"output-mode", "m",
+		"terminal",
+		"The trace output mode, possibles values are terminal (default) and seccomp-profile.")
+	seccompAdvisorStartCmd.PersistentFlags().StringVar(&seccompProfileName,
+		"seccomp-profile-name", "",
+		"The seccomp profile which will be created when using seccomp-profile as value for --output-mode.")
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorStopCmd)
+	seccompAdvisorCmd.AddCommand(seccompAdvisorListCmd)
+}
+
+func outputModeToTraceOutputMode(outputMode string) (string, error) {
+	switch outputMode {
+	case "terminal":
+		return "Status", nil
+	case "seccomp-profile":
+		return "ExternalResource", nil
+	default:
+		return "", fmt.Errorf("%q is not an accepted value for --output-mode, possible values are: terminal (default) and seccomp-profile", outputMode)
+	}
+}
+
+// runSeccompAdvisorStart starts monitoring of syscalls for the given
+// parameters.
+func runSeccompAdvisorStart(cmd *cobra.Command, args []string) error {
+	if params.Podname == "" {
+		return errors.New("Usage: kubectl gadget seccompadvisor start -p podname")
+	}
+
+	traceOutputMode, err := outputModeToTraceOutputMode(outputMode)
+	if err != nil {
+		return err
+	}
+
+	if traceOutputMode != "ExternalResource" && seccompProfileName != "" {
+		return errors.New("You can only use --seccomp-profile-name with --output seccomp-profile.")
+	}
+
+	if traceOutputMode == "ExternalResource" && seccompProfileName == "" {
+		return errors.New("You must specify --seccomp-profile-name when using --output seccomp-profile.")
+	}
+
+	config := &utils.TraceConfig{
+		GadgetName:        "seccomp",
+		Operation:         "start",
+		TraceOutputMode:   traceOutputMode,
+		TraceOutput:       seccompProfileName,
+		TraceInitialState: "Started",
+		CommonFlags:       &params,
+	}
+
+	traceID, err := utils.CreateTrace(config)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("%s\n", traceID)
+
+	return nil
+}
+
+// runSeccompAdvisorStop reports an already running trace which ID was given
+// as parameter.
+func runSeccompAdvisorStop(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("Usage: kubectl gadget seccomp-advisor stop global-trace-id\n")
+	}
+
+	callback := func(results []gadgetv1alpha1.Trace) error {
+		for _, i := range results {
+			if i.Spec.OutputMode == "ExternalResource" {
+				fmt.Printf("Successfully created seccomp profile %q!\n", i.Spec.Output)
+
+				return nil
+			}
+
+			fmt.Printf("%v\n", i.Status.Output)
```

Comment:
I think we should not print this if the `i.Status.Output` is empty so that we don't print unnecessary new lines.